### PR TITLE
feat(providers): support Vercel AI Gateway embeddings and images

### PR DESCRIPTION
**Pull request by newnol**
## Description

This change adds support for using Vercel AI Gateway for embeddings and image generation, extending the existing provider integration beyond text/chat completion workflows.

With this update, users can route embedding and image requests through Vercel AI Gateway using the same provider configuration, making it possible to access more model capabilities through a single unified gateway.

## Changes

- Added Vercel AI Gateway support for embedding requests.
- Added Vercel AI Gateway support for image generation requests.
- Reused the existing provider configuration flow for API keys and base URL handling.
- Kept existing chat/completion behavior unchanged.

## Motivation

Vercel AI Gateway supports multiple model families and capabilities behind a unified API. Supporting embeddings and images makes the provider integration more complete and reduces the need for separate provider-specific configuration.

## Test Plan

- Verified that embedding requests can be routed through Vercel AI Gateway.
- Verified that image generation requests can use Vercel AI Gateway provider settings.
- Confirmed existing provider behavior remains compatible.

### CODE REVIEW

**Review by tonydwb — APPROVED** [review]
## Code Review Summary

**Verdict: Approved**

Extends the existing Vercel AI Gateway provider support with robust upgrade paths:

- Embeddings endpoint (`/v1/embeddings`) and image generation (`/v1/images/generations`) are now wired through the OpenAI-compatible adapter stack, matching the existing chat-completion path.
- Provider config gains `thinkingConfig`, `modelsFetcher`, and added service kinds (`embedding`, `image`, `imageToText`, `webSearch`), plus 429 retry for chat.
- A dedicated usage handler (`/v1/credits`) surfaces USD credit balance in the quota UI, with sane fallback for BYOK/unfunded accounts.
- Claude thinking-to-reasoning_effort mapping added so encryption-format providers preserve reasoning intent.
- New tests cover embeddings and image-generation request construction.

### ✅ Looks Good
- Consistent adapter registration pattern (constants + endpoint + tests).
- Quota integration handles edge cases (divide-by-zero, no allocation).
- Existing provider datasets now carry the right metadata shape.

**Comment by newnol** [comment]
CleanShot 2026-06-06 at 20 18 53@2x
Hi maintainers,

I added a screenshot of the updated Vercel AI Gateway quota display.

The Quota Tracker now shows both Used (USD) and Remaining (USD), with the remaining value displayed as `remaining / total`. I tested it with multiple Vercel AI Gateway accounts, and the values/percentages look correct.

This should make it easier for users to monitor their Vercel credits directly inside 9Router.

Thanks!

File: open-sse/services/usage.js [hunk 1]
```diff
@@ -1152,6 +1159,89 @@ async function getMiniMaxUsage(apiKey, provider, proxyOptions = null) {
...
+    const data = await response.json();
+
+    // Vercel returns numeric strings; coerce safely.
+    const balance = Number(data?.balance) || 0;
+    const totalUsed = Number(data?.total_used) || 0;
+
+    // Vercel gives $5/month free credit. The API doesn't return the
+    // monthly allocation so we use the known constant as the denominator.
+    const MONTHLY_CREDIT = 5;
+    const remainingPercentage = (balance / MONTHLY_CREDIT) * 100;
```
[the remark this answers is not in the document]
Comment (newnol) [reply open-sse/services/usage.js:1209]: This because mostly people using free tier from Vercel that provide 5$, so i hard-codes it

**Comment by newnol** [comment]
Closing to reopen from a dedicated feature branch instead of master.
